packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -227,7 +227,10 @@ const _Dropdown = ({
           return;
         }
 
-        const isOutsideClick = !dropdown.contains(target) && !isTagDismissedRef.current?.value;
+        const isOutsideClick =
+          !dropdown.contains(target) &&
+          !isTagDismissedRef.current?.value &&
+          document.body.contains(target);
```

Comment:
isOutsideClick was getting true when icon was clicked because when you click ChevronDown icon, it opens dropdown and switches to ChevronUp icon. So ChevronDown icon is the `target` here which did not exist in DOM after click so `dropdown.contains(target)` was returning true.

Now we also check if target exists in DOM with `document.body.contains(target)`

Reply:
How was this working before?

Reply:
No clue lmao. I think the sequence of state setting changed after `useControllableState` implementation. Earlier it must've been calling set state twice but maybe in this sequence 
- Close dropdown
- Open dropdown

But now it seems to call
- Open dropdown
- Close dropdown

so its closing the dropdown immediately after open when icon of DropdownLink is clicked